plugins/chat/spec/services/update_channel_spec.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+RSpec.describe Chat::Service::UpdateChannel do
+  subject(:result) { described_class.call(guardian: guardian, channel_id: channel.id, **params) }
+
+  fab!(:channel) { Fabricate(:chat_channel) }
+  fab!(:current_user) { Fabricate(:admin) }
+
+  let(:guardian) { Guardian.new(current_user) }
+  let(:params) do
+    {
+      name: "cool channel",
+      description: "a channel description",
+      slug: "snail",
+      allow_channel_wide_mentions: true,
+      auto_join_users: false,
+    }
+  end
+
+  context "when the user cannot edit the channel" do
+    fab!(:current_user) { Fabricate(:user) }
+
+    it { is_expected.to fail_a_policy(:check_channel_permission) }
+  end
+
+  context "when the user tries to edit a DM channel" do
+    fab!(:channel) { Fabricate(:direct_message_channel, users: [current_user, Fabricate(:user)]) }
+
+    it { is_expected.to fail_a_policy(:no_direct_message_channel) }
+  end
+
+  context "when channel is a category one" do
+    context "when a valid user provides valid params" do
+      let(:message) do
+        MessageBus.track_publish(ChatPublisher::CHANNEL_EDITS_MESSAGE_BUS_CHANNEL) { result }.first
```

Comment:
Not to be addressed in that PR, but we've had some flaky test whenever two pieces of code would publish a message and would break the specs that are looking for the `first` message in the bus. Do/should we have a way to filter for a certain type of message and then do `.first` on it? 🤔

Reply:
yes agreed, generally I think `include?` on a map of events is just enough